File: 0171-excel-sheet-column-number/0171-excel-sheet-column-number.rb
# @param {String} column_title
# @return {Integer}
def title_to_number(column_title)
  codes = []
  column_title.each_char{ |ch| codes << ch.ord-'A'.ord+1 }
  n = ans = 0
  popped = nil
  while !codes.empty?
    popped = codes.pop
    ans += popped * 26 ** n
    n += 1
  end
  ans
end

File: 0171-excel-sheet-column-number/0171-excel-sheet-column-number.py
class Solution:
  def titleToNumber(self, columnTitle: str) -> int:
    codes = [ord(ch)-ord('A')+1 for ch in columnTitle]
    n = ans = 0
    popped = None
    while codes:
      popped = codes.pop()
      ans += popped * pow(26, n)
      n += 1
    return ans